src/lib.rs

```diff
@@ -6,12 +6,39 @@
 
 use once_cell::sync::Lazy;
 
+#[cfg(feature = "kv")]
+use slog::KV;
 use tracing_core::{
     callsite, dispatcher, field, identify_callsite,
     metadata::{Kind, Level},
     subscriber, Event, Metadata,
 };
 
+#[cfg(feature = "kv")]
+/// An allocating serializer to use for serializing key-value pairs in a [`slog::Record`]
+#[derive(Default)]
+struct TracingKvSerializer {
+    storage: String,
+}
+
+#[cfg(feature = "kv")]
+impl TracingKvSerializer {
+    fn as_str(&self) -> &str {
+        self.storage
+            .get(..self.storage.len().saturating_sub(1))
+            .unwrap_or_default()
+    }
+}
+
+#[cfg(feature = "kv")]
+impl slog::Serializer for TracingKvSerializer {
+    fn emit_arguments(&mut self, key: slog::Key, val: &core::fmt::Arguments) -> slog::Result {
+        self.storage
+            .push_str(&format_args!("{key}={val},").to_string());
```

Comment:
Is the to_string() is needed here?
```suggestion
            .push_str(&format!("{key}={val},"));
```